fu/basic/Fu.py

```diff
@@ -105,6 +107,8 @@ def update_vector_factor_counter():
                                                                      VectorFactorType))
           elif s.recv_opt.msg.is_last_ctrl & s.reached_vector_factor:
             s.vector_factor_counter <<= 0
+          else:
+            s.vector_factor_counter <<= s.vector_factor_counter
```

Comment:
For `<<=` in `@update_ff`, i.e., register update (rather than wire update), we don't need initialization of `<<= 0` and `else`. In other words, there is no latch for register.